src/PluginsManager/Microsoft.Performance.Toolkit.Plugins.Runtime/Result/InstalledPluginsResults.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Performance.Toolkit.Plugins.Runtime.Result
+{
+    public class InstalledPluginsResults
+    {
+        public InstalledPluginsResults(
+                        IReadOnlyCollection<InstalledPlugin> validPlugins,
+            IReadOnlyCollection<InstalledPluginInfo> invalidPluginsInfo)
```

Comment:
How does this work exactly? If there's some problem fetching info for a plugin, how do we have enough information to create an `InstalledPluginInfo` for it?

Reply:
This is used only by `GetInstalledPluginsAsync` which does three things: 1. Read all records from the plugin registry  2. For each record, read their metadata from where they're installed. 3. For each record, create a `InstalledPlugin` instance and return all the created `InstalledPlugin`s. 

This method will throw `PluginRegistryException`s  but will swallow exceptions thrown by individual plugin records when reading the metadata.  For those that fail to read metadata, they are added to `InvalidPluginsInfo` in this result class (they will always have `InstalledPluginInfo`) so that the client is aware of these plugins.

Reply:
Got it. For some reason, I assumed the metadata was also on InstalledPluginInfo